src/client/client.c

```diff
@@ -304,12 +211,24 @@ int client_call_manager(Client *client) {
 
         if (streq(client->op, "help")) {
                 r = print_client_usage("hirte");
-        } else if (streq(client->op, "list-units")) {
-                if (client->opargc == 0) {
-                        r = method_list_units_on_all(client);
-                } else {
-                        r = method_list_units_on(client->opargv[0], client);
+        } else if (streq(client->op, CMD_LIST_UNITS)) {
+                char *glob_filter = NULL;
+                int r = parse_list_units_options(org_argc, org_argv, &glob_filter);
```

Comment:
I think since we are already parsing command line options in main.c, we should do that there as well and configure the client object from there.

Reply:
I also don't like locating it here in `clien.c`. However, the `--filter` is specific for the `list-units` commands (for now) and, therefore, it feels off to parse this option in a general way in `main.c`. 
Ideally, we would've some sort of command builder, e.g. detecting the first parameter (which is the command such as `list-units`, `start`, etc.) and then pass all options down to the builder - only extracting and using those that are needed.

Reply:
Yeah, but as long as we don't have that - I think tossing everything to the same place - parameter parsing, execution, output - makes the code very messy, hard to understand and hard to change.

Reply:
I missed that configuration object is not used in hirtectl, please ignore my comment

Reply:
We have discussed "in person" and agreed that for now we will leave parsing where it is (in `client.c`), and we will later introduce some sort of command builder that will address this, and also will allow displaying help for each method separately